File: src/test/fixtures/typescript/anonymousCallables.ts
function named(value: number): number {
    return value;
}

export const moduleArrow = (left: number, right: number): number => left + right;
const moduleFunction = function (value: number): number {
    return value + 1;
};

class Operations {
    classBound = (value: number): number => value + 1;

    namedMethod(value: number): number {
        return value;
    }
}

function outer(items: number[]): number[] {
    const localBound = (value: number): number => value * 2;
    return items.map(item => item + 1);
}

File: src/test/fixtures/csharp/AnonymousCallables.cs
using System;

class Operations
{
    static Func<int, int, int> ClassBound = (left, right) => left + right;

    static Predicate<int> ClassMethod = delegate(int value)
    {
        return value > 0;
    };

    int NamedMethod(int value)
    {
        return value;
    }

    void Outer(int[] items)
    {
        Func<int, int> localBound = value => value * 2;
        Array.ConvertAll(items, item => item + 1);
    }
}
